File: migration-backend/pkg/handler/likenft/model/migration.go
package model

import "time"

type Migration struct {
	Status  MigrationStatus `json:"status"`
	Classes []Class         `json:"classes"`
	NFTs    []NFT           `json:"nfts"`
}

type MigrationStatus string

const (
	MigrationStatusPreview    MigrationStatus = "preview"
	MigrationStatusPending    MigrationStatus = "pending"
	MigrationStatusInProgress MigrationStatus = "in_progress"
	MigrationStatusSuccess    MigrationStatus = "success"
	MigrationStatusFailed     MigrationStatus = "failed"
)

type Class struct {
	Id           string               `json:"id"`
	Name         string               `json:"name"`
	LikerlandUrl string               `json:"likerland_url"`
	Image        string               `json:"image"`
	Status       ClassMigrationStatus `json:"status"`
	EnqueuedTime time.Time            `json:"enqueued_time,omitempty"`
	EVMTxHash    string               `json:"evm_tx_hash,omitempty"`
	FailedReason string               `json:"failed_reason,omitempty"`
}

type ClassMigrationStatus string

const (
	ClassMigrationStatusPending    ClassMigrationStatus = "pending"
	ClassMigrationStatusScheduled  ClassMigrationStatus = "scheduled"
	ClassMigrationStatusInProgress ClassMigrationStatus = "in_progress"
	ClassMigrationStatusSuccess    ClassMigrationStatus = "success"
	ClassMigrationStatusFailed     ClassMigrationStatus = "failed"
)

type NFT struct {
	ClassId      string             `json:"class_id"`
	Id           string             `json:"id"`
	Name         string             `json:"name"`
	LikerlandUrl string             `json:"likerland_url"`
	Image        string             `json:"image"`
	Status       NFTMigrationStatus `json:"status"`
	EnqueuedTime time.Time          `json:"enqueued_time,omitempty"`
	EVMTxHash    string             `json:"evm_tx_hash,omitempty"`
	FailedReason string             `json:"failed_reason,omitempty"`
}

type NFTMigrationStatus string

const (
	NFTMigrationStatusPending    NFTMigrationStatus = "pending"
	NFTMigrationStatusScheduled  NFTMigrationStatus = "scheduled"
	NFTMigrationStatusInProgress NFTMigrationStatus = "in_progress"
	NFTMigrationStatusSuccess    NFTMigrationStatus = "success"
	NFTMigrationStatusFailed     NFTMigrationStatus = "failed"
)

File: likenft-migration/apis/models/migration.ts
import { z } from 'zod';

export const MigrationClassStatusSchema = z.enum([
  'pending',
  'scheduled',
  'in_progress',
  'success',
  'failed',
]);

export const MigrationClassSchema = z.object({
  id: z.string(),
  name: z.string(),
  likerland_url: z.string(),
  image: z.string(),
  status: MigrationClassStatusSchema,
  enqueued_time: z.string().nullable(),
  evm_tx_hash: z.string().nullable(),
  failed_reason: z.string().nullable(),
});

export const MigrationNFTStatusSchema = z.enum([
  'pending',
  'scheduled',
  'in_progress',
  'success',
  'failed',
]);

export const MigrationNFTSchema = z.object({
  class_id: z.string(),
  id: z.string(),
  name: z.string(),
  likerland_url: z.string(),
  image: z.string(),
  status: MigrationNFTStatusSchema,
  enqueued_time: z.string().nullable(),
  evm_tx_hash: z.string().nullable(),
  failed_reason: z.string().nullable(),
});

export const MigrationStatusSchema = z.enum([
  'preview',
  'pending',
  'in_progress',
  'success',
  'failed',
]);

export const MigrationSchema = z.object({
  status: MigrationStatusSchema,
  classes: z.array(MigrationClassSchema),
  nfts: z.array(MigrationNFTSchema),
});

export type Migration = z.infer<typeof MigrationSchema>;
